### DISCUSSION TITLE
Feedback Provider API exposing multiple errorRecords and accepting multiple Recommendations

### DISCUSSION THREAD

**Discussion by JustinGrote** [PowerShell/PowerShell#20834] [Enhancements]
@StevenBucher98 wanted to start this as a discussion before making it an issue.

One particular scenario I've run into is when someone is not just running a simple command, they are running a script or whatnot that may issue multiple nonterminating errors.

In such a scenario, I want to be able to analyze the multiple errors and provide a recommendation for each. I cannot do this with the `LastError` context interface.

As such, as a non-breaking change I would like to request a new interface for the feedbackprovider that provides a `LastErrors` property for all errors that occured in the last executed command, as well as accept an array of `RecommendedAction` objects. This could ultimately become the primary feedback provider interface, since acting on the last error and returning only one recommendedAction could be adapted to this interface.

In addition, it seems the lines between RecommendedAction and ScriptAnalyzer Analyzers are very very close, it may be worth unifying some interfaces there to make them compatible with each other, especially to potentially offer a "quick fix" option to the command line via a feedbackprovider that will rewrite the command for the user if a certain keybinding is pressed for instance.

Thoughts?

**Comment by StevenBucher98** [comment]
Really interesting idea, feedback provider was primarily designed for single command executions in the interactive terminal so I imagine just the last error is considered there. I agree this is starting to blur the lines a little with PSSA rules and feedback providers, definitely need to think about unification of interface. 

cc @daxian-dbw for your thoughts

**Reply by JustinGrote** [reply]
Simple example: 
`get-childitem |% { Write-Error 'SOmething went wrong'}`

With current model I can only recommend on the last file.
